client/post-editor/editor-drawer/taxonomies.jsx

```diff
@@ -24,35 +25,26 @@ function EditorDrawerTaxonomies( { siteId, postType, postTerms, taxonomies } ) {
 				<QueryTaxonomies { ...{ siteId, postType } } />
 			) }
 			{ map( taxonomies, ( taxonomy ) => {
-				if ( 'post_format' === taxonomy.name ) {
+				const { name, label, hierarchical } = taxonomy;
+				if ( 'post_format' === name ) {
 					// Post format has its own dedicated accordion
 					return;
 				}
 
-				if ( taxonomy.hierarchical ) {
-					return (
-						<Accordion
-							key={ taxonomy.name }
-							title={ taxonomy.label }
-							icon={ <Gridicon icon="category" /> }
-						>
-							Work in Progress
-						</Accordion>
-					);
-				} else {
-					return (
-						<Accordion
-							key={ taxonomy.name }
-							title={ taxonomy.label }
-							icon={ <Gridicon icon="tag" /> }
-						>
-							<TermTokenField
-								postTerms={ postTerms }
-								taxonomyName={ taxonomy.name }
-							/>
-						</Accordion>
-					);
-				}
+				const icon = hierarchical ? 'folder' : 'tag';
+
+				return (
+					<Accordion
+						key={ name }
+						title={ label }
+						icon={ <Gridicon icon={ icon } /> }
+					>
+					{ hierarchical
```

Comment:
Now I'm just being obnoxious, but because the components take the same props, you could assign the component to be rendered into a variable:

```diff
diff --git a/client/post-editor/editor-drawer/taxonomies.jsx b/client/post-editor/editor-drawer/taxonomies.jsx
index fd1f0f8..0d1fc04 100644
--- a/client/post-editor/editor-drawer/taxonomies.jsx
+++ b/client/post-editor/editor-drawer/taxonomies.jsx
@@ -32,6 +32,7 @@ function EditorDrawerTaxonomies( { siteId, postType, postTerms, taxonomies } ) {
                                }
 
                                const icon = hierarchical ? 'folder' : 'tag';
+                               const Selector = hierarchical ? TermSelector : TermTokenField;
 
                                return (
                                        <Accordion
@@ -39,10 +40,7 @@ function EditorDrawerTaxonomies( { siteId, postType, postTerms, taxonomies } ) {
                                                title={ label }
                                                icon={ <Gridicon icon={ icon } /> }
                                        >
-                                       { hierarchical
-                                               ? <TermSelector postTerms={ postTerms } taxonomyName={ name } />
-                                               : <TermTokenField postTerms={ postTerms } taxonomyName={ name } />
-                                       }
+                                               <Selector postTerms={ postTerms } taxonomyName={ name } />
                                        </Accordion>
                                );
                        } ).filter( Boolean ) }
```